File: app/src/main/java/com/silentgames/silent_planet/model/fractions/factionType/Pirates.java
package com.silentgames.silent_planet.model.fractions.factionType;

import com.silentgames.silent_planet.model.entities.ground.fractions.Pirate;
import com.silentgames.silent_planet.model.fractions.Fractions;
import com.silentgames.silent_planet.model.fractions.FractionsEnum;

/**
 * Created by gidroshvandel on 27.09.16.
 */
public class Pirates extends Fractions {

    private static Pirates instance;

    public static synchronized Pirates getInstance(){
        if(instance == null){
            instance = new Pirates();
        }
        return instance;
    }

    private Pirates() {
        this.setFractionsEnum(FractionsEnum.Pirates);
    }
}

File: app/src/main/java/com/silentgames/silent_planet/model/fractions/factionType/Pirates.kt
package com.silentgames.silent_planet.model.fractions.factionType

import com.silentgames.silent_planet.model.fractions.Fractions
import com.silentgames.silent_planet.model.fractions.FractionsType

/**
 * Created by gidroshvandel on 27.09.16.
 */
class Pirates private constructor() : Fractions(FractionsType.PIRATE) {

    companion object {

        private var instance: Pirates? = null

        @Synchronized
        fun getInstance(): Pirates {
            if (instance == null) {
                instance = Pirates()
            }
            return instance!!
        }
    }
}
